test/e2e/generator/random.go

```diff
@@ -83,3 +83,25 @@ func (usc uniformSetChoice) Choose(r *rand.Rand) []string {
 	}
 	return choices
 }
+
+// weightedChoice chooses a single random key from a map of keys and weights.
+type weightedChoice map[interface{}]uint
+
+func (wc weightedChoice) Choose(r *rand.Rand) interface{} {
```

Comment:
Resurrected from v0.34.x. Using weighted choice allows us to only set a proportion of the nodes to use the alternate version. The other option, uniform-choice, would lead to 50% of the nodes being the alternate version which is clearly more than needed to simply test that new code is compatible with old nodes.

Reply:
Nice!